[b-7] currency conversion

Jokeroff opened the pull request.

Jokeroff committed:
[fixes for previous tasks]:
- DataSource changed
- SortParams params  instead of Map <String, String> params
- MovieParamsValidator totally changed

Jokeroff committed:
[b-7]: currency conversion draft

Jokeroff committed:
[b-7]:currency conversion
- validation for parameter added
- static methods
- get single CurrencyEntity

Ramoid commented on src/main/java/com/lebediev/movieland/service/conversion/CachedCurrency.java:
```diff
@@ -0,0 +1,105 @@
+        List<CurrencyEntity> currencyEntityList = new ArrayList<>();
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode rootNode = objectMapper.readTree(json);
```
можно вместо получения стринга в getCurrenciesFromUrl через цепочку  BufferedReader  InputStreamReader напрямую заюзать
objectMapper.readTree(InputStream)

Jokeroff replied:
не знал, спасибо
готово

Ramoid commented on src/main/java/com/lebediev/movieland/service/conversion/CurrencyConverter.java:
```diff
@@ -0,0 +1,24 @@
+        long startTime = System.currentTimeMillis();
+        CurrencyEntity currencyEntity = cachedCurrency.getCurrencyEntity(currency);
+        double rate = currencyEntity.getRate();
+        movie.setPrice(movie.getPrice()/rate);
```
Округление с BigDecimal

Jokeroff replied:
done, thank you!

Jokeroff committed:
[b-7]: fixes
- objectMapper.readTree(reader);
- isExpired + test
- tests added: for converter, validator, controller
- rounding added

juliamanunia commented on src/main/java/com/lebediev/movieland/service/conversion/CachedCurrency.java:
```diff
@@ -0,0 +1,93 @@
+        }
+    }
+
+    public static boolean isExpired(CurrencyEntity currencyEntity){
```
url returns CCYs for current day, do you really want to check it for expiration?

Jokeroff replied:
for case when we could not obtain actual data and our cache will contain expired rates

juliamanunia commented on src/main/java/com/lebediev/movieland/service/conversion/CachedCurrency.java:
```diff
@@ -0,0 +1,93 @@
+    }
+
+
+    public static List<CurrencyEntity> getCurrenciesFromUrl(String url){
```
why public?

Jokeroff replied:
thank you, changed. I had plans to test it somehow )

juliamanunia commented on src/main/java/com/lebediev/movieland/service/conversion/CachedCurrency.java:
```diff
@@ -0,0 +1,93 @@
+
+@Service
+public class CachedCurrency {
+    private final static Logger log = LoggerFactory.getLogger(CachedCurrency.class);
```
if static, should be upper case

juliamanunia commented on src/main/java/com/lebediev/movieland/service/conversion/CachedCurrency.java:
```diff
@@ -0,0 +1,93 @@
+public class CachedCurrency {
+    private final static Logger log = LoggerFactory.getLogger(CachedCurrency.class);
+    private volatile List<CurrencyEntity> currencyList;
+    private final static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
```
upper case

Jokeroff committed:
[b-7]: currency conversion minor fixes

Jokeroff committed:
[b-7] minor fixes:
- properties separate folder
- CachedCurrency changes
